collectors/ebpf.plugin/ebpf_mount.c

```diff
@@ -0,0 +1,257 @@
+
+static int read_thread_closed = 1;
+static netdata_idx_t *mount_values = NULL;
+
+static struct bpf_link **probe_links = NULL;
+static struct bpf_object *objects = NULL;
+
+static netdata_idx_t mount_hash_values[NETDATA_MOUNT_END];
+
+struct netdata_static_thread mount_threads = {"MOUNT KERNEL",
+                                              NULL, NULL, 1, NULL,
+                                              NULL,  NULL};
+
+/*****************************************************************
+ *
+ *  FUNCTIONS TO CLOSE THE THREAD
+ *
+ *****************************************************************/
+
+/**
+ * Clean up the main thread.
+ *
+ * @param ptr thread data.
+ */
+static void ebpf_mount_cleanup(void *ptr)
+{
+    ebpf_module_t *em = (ebpf_module_t *)ptr;
+    if (!em->enabled)
+        return;
+
+    freez(mount_threads.thread);
+    freez(mount_values);
+
+    if (probe_links) {
+        struct bpf_program *prog;
+        size_t i = 0 ;
+        bpf_object__for_each_program(prog, objects) {
+            bpf_link__destroy(probe_links[i]);
+            i++;
+        }
+        bpf_object__close(objects);
+    }
+}
+
+/*****************************************************************
+ *
+ *  MAIN LOOP
+ *
+ *****************************************************************/
+
+/**
+ * Read global table
+ *
+ * Read the table with number of calls for all functions
+ */
+static void read_global_table()
+{
+    uint32_t idx;
+    netdata_idx_t *val = mount_hash_values;
+    netdata_idx_t *stored = mount_values;
+    int fd = mount_maps[NETDATA_KEY_MOUNT_TABLE].map_fd;
+
+    for (idx = NETDATA_KEY_MOUNT_CALL; idx < NETDATA_MOUNT_END; idx++) {
+        if (!bpf_map_lookup_elem(fd, &idx, stored)) {
+            int i;
+            int end = ebpf_nprocs;
+            netdata_idx_t total = 0;
+            for (i = 0; i < end; i++)
+                total += stored[i];
+
+            val[idx] = total;
+        }
+    }
+}
+
+/**
+ * Mount read hash
+ *
+ * This is the thread callback.
+ * This thread is necessary, because we cannot freeze the whole plugin to read the data.
+ *
+ * @param ptr It is a NULL value for this thread.
+ *
+ * @return It always returns NULL.
+ */
+void *ebpf_mount_read_hash(void *ptr)
+{
+    read_thread_closed = 0;
+
+    heartbeat_t hb;
+    heartbeat_init(&hb);
+
+    ebpf_module_t *em = (ebpf_module_t *)ptr;
+
+    usec_t step = NETDATA_LATENCY_MOUNT_SLEEP_MS * em->update_time;
+    while (!close_ebpf_plugin) {
+        usec_t dt = heartbeat_next(&hb, step);
+        (void)dt;
+
+        read_global_table();
+    }
+    read_thread_closed = 1;
+
+    return NULL;
+}
+
+/**
+ * Send data to Netdata calling auxiliar functions.
+ *
+ * @param em the structure with thread information
+*/
+static void ebpf_mount_send_data()
+{
+    int i, j;
+    int end = NETDATA_EBPF_MOUNT_SYSCALL;
+    for (i = NETDATA_KEY_MOUNT_CALL, j = NETDATA_KEY_MOUNT_ERROR; i < end; i++, j++) {
```

Comment:
Maybe just my opinion but, this loop is really hard to read, even though it's doing something simple. If it was just looping through an enum from start to end as in `for (i = NETDATA_KEY_MOUNT_CALL; i < NETDATA_MOUNT_END; i++)`, it would be alright, but here it's confusing.

Can we just unroll the loop and inline it with explicit enum values? e.g.

```c
// update call count
mount_publish_aggregated[NETDATA_KEY_MOUNT_CALL].ncall = mount_hash_values[NETDATA_KEY_MOUNT_CALL];
mount_publish_aggregated[NETDATA_KEY_UMOUNT_CALL].ncall = mount_hash_values[NETDATA_KEY_UMOUNT_CALL];

// update err count
mount_publish_aggregated[NETDATA_KEY_MOUNT_CALL].nerr = mount_hash_values[NETDATA_KEY_MOUNT_ERROR];
mount_publish_aggregated[NETDATA_KEY_UMOUNT_CALL].nerr = mount_hash_values[NETDATA_KEY_UMOUNT_ERROR];
```

Reply:
I understand your point of view, but other threads that have more than two dimensions are using loop, so I brought loop here to keep one unique pattern.

I will give a suggestion, for this specific PR we will keep this pattern, but let us make another PR doing something like `go.d.plugin`, where there is a specific file describing how to test and develop with some `module`. I do not want to have different patterns in different threads to avoid confusion.